### ISSUE TITLE
A little bug in scapture_quant

### ISSUE THREAD

**Issue by YangShi-PKU** [YangLab/SCAPTURE#8]
Hi guowei,
When I run `scapture -m PASquant` with the parameter `--celllist FromBAM` there is a "File not found" bug. This is caused by an unexpected location of BAM file generated by featurecounts.
Suppose my BAMFile is /bamdir/P01.bam, PREFIX is P01.PASquant, working directory is /resdir/:
In scapture_quant, line 127:
`featureCounts -M -O --largestOverlap -F GTF -t exon -g gene_id -s 1 -T 4 -R BAM -a $PREFIX".KeepPAS.gtf" -o $PREFIX".KeepCell.assigned" $BAMFile &> /dev/null`
will generate a file named "P01.bam.featureCounts.bam" in /resdir, but in line 128:
`samtools sort -@ 4 $BAMFile".featureCounts.bam" -o $PREFIX".KeepCell.reassigned.bam" &> /dev/null`
it tries to find this file using path /bamdir/P01.bam.featureCounts.bam.

I solved it by adding `--Rpath` to line 127 to specify the output directory of *featureCounts.bam. Maybe you have better solution and can fix it in next version of SCAPTURE. 

Best wishes,
Yang

**Comment by liguowei-CAS** [comment]
Thanks for pointing this out! It should be noted that the original BAM file includes numerous barcodes, lots of which were not real cell barcodes. Using "--celllist FromBAM" means _all barcodes_ in the BAM file will be counted in umitool, and it might take a lot of time.

**Comment by liguowei-CAS** [comment]
Thanks for pointing this out! It should be noted that the original BAM file includes numerous barcodes, lots of which were not real cell barcodes. Using "--celllist FromBAM" means _all barcodes_ in the BAM file will be counted in umitool, and it might take a lot of time.

**Comment by YangShi-PKU** [comment]
Hi guowei,
Thanks for your suggestion. I've filtered the Cellranger BAM at the beginning. Thus "unfortunately" used --celllist FromBAM and ran into this problem. You are absolutely right that --celllist $CellbarcodeFile is a better choice in most cases.

Yang

**Comment by YangShi-PKU** [comment]
Oh I just remembered that, after "FromBAM" gave me an error, I tried to use--celllist $CellbarcodeFile. However then FilterBamByTag throw out an error. It seems my BAM header has some issue....although I found no difference from the initial header. So I turned to "FromBAM" again.
I just want to complain that those tools from Broad Institute always require some useless things like the header, readgroup chromosome order, etc. 
Anyway, thanks a lot!

Best
Yang
